### PR TITLE
ET-4300 Add ort

### PR DESCRIPTION
**Pull request by acrrd**
**Reference**

- [ET-4300]
- [ET-4912]
- [ET-4301]
- followed by #1066

**Summary**

Initial support for ort to compute the embeddings.
It can be selected with an option.
Some data structure it is converted to the tract one to avoid changing too much.

To deploy it, some changes can be needed in a follow-up.

Once we have ort working and deployed, we can remove tract.

- currently the onnxruntime lib is added to the assets bucket, probably will be added to the ci image later (but no gpu lib yet)
- integration tests already run with onnxruntime instead of tract
- engine defaults run with onnxruntime
- webservice build also copies onnxruntime files
- update the benchmark, the results are in [ET-4301]
- image builds successfully: https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/5857592118

[ET-4300]: https://xainag.atlassian.net/browse/ET-4300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ET-4912]: https://xainag.atlassian.net/browse/ET-4912?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ET-4301]: https://xainag.atlassian.net/browse/ET-4301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-4301]: https://xainag.atlassian.net/browse/ET-4301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -151,11 +151,11 @@ web-dev-up:
         ln -s "./assets/xaynia_v0002" "./web-api/assets"
     fi
     export HOST_PORT_SCOPE=30
-    docker-compose -p "$PROJECT" -f "./web-api/compose.db.yml" up --detach --remove-orphans --build
+    podman-compose -p "$PROJECT" -f "./web-api/compose.db.yml" up --detach --remove-orphans --build
```
Comment (rustonaut) [inline justfile:154]: :wrench:  We decided to always use docker due to the huge amount of problems we had caused by subtle differences between both of them. I think we might also be using some options podman doesn't support and silently ignores.

Comment (acrrd) [reply justfile:154]: Done

File: .github/workflows/web-service.yml [hunk 2]
```diff
@@ -4,26 +4,40 @@ on:
...
           - arm64
           - amd64
+        default: arm64
+        required: true
       tag:
-        type: string
         description: >
           A tag for image identification. This override the default.
-          Default is <branch>-<timestamp>-<commit hash>-<model_name>_<model_version>-<platform>.
+          Default is <branch>-<timestamp>-<commit hash>-<model_name>_<model_version>-<runtime_name>_<runtime_version>-<platform>.
```
Comment (acrrd) [inline .github/workflows/web-service.yml:37]: I'm not sure why we need this, the code can switch between runtime with a configuration options and the ort version can be take from the commit.

Comment (acrrd) [reply .github/workflows/web-service.yml:37]: The space for the tag is limited so I don't want to overload it with info that we can easily get somewhere else

File: .github/workflows/web-service.yml [hunk 3]
```diff
@@ -4,26 +4,40 @@ on:
...
+      runtime_name:
+        description: Name of the runtime
+        type: string
+        default: ort
+        required: true
+      runtime_version:
+        description: Version of the runtime
+        type: string
+        default: v1.15.1
+        required: true
```
[the remark this answers is not in the document]
Comment (janpetschexain) [reply .github/workflows/web-service.yml]: i moved it into the github env context. it's kinda difficult to read from somewhere and will only change if the underlying version in the code changes.

File: .github/workflows/web-service.yml (hunk 2 above)
Comment (janpetschexain) [reply .github/workflows/web-service.yml]: i removed it.

**Comment by janpetschexain** [comment]
rebased and resolved merge conflicts, pls have a look @acrrd

**Review by rustonaut — APPROVED** [review]
lgtm

**Pull request merged**